rq/queue.py

```diff
@@ -289,15 +289,22 @@ def enqueue_dependents(self, job):
         # TODO: can probably be pipelined
         from .registry import DeferredJobRegistry
 
-        registry = DeferredJobRegistry(self.name, self.connection)
+
 
         while True:
             job_id = as_text(self.connection.spop(job.dependents_key))
             if job_id is None:
                 break
             dependent = self.job_class.fetch(job_id, connection=self.connection)
-            registry.remove(dependent)
-            self.enqueue_job(dependent)
+            registry = DeferredJobRegistry(dependent.origin, self.connection)
+            with self.connection._pipeline() as pipeline:
+                registry.remove(dependent, pipeline=pipeline)
+                if dependent.origin == self.name:
+                    self.enqueue_job(dependent)
```

Comment:
Thanks for the update! Do you mind adding a `pipeline` argument to `enqueue_job` so we can do all these commands in one network call?

Thanks again :)